internal/storage/vmbus/vmbus.go

```diff
@@ -0,0 +1,43 @@
+// +build linux
+
+package vmbus
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+var Filepathglob = filepath.Glob
+
+// WaitForVMBusDevicePath waits for the vmbus device to exist at /sys/bus/vmbus/devices/<vmbusGUIDPattern>...
+func WaitForVMBusDevicePath(ctx context.Context, vmbusGUIDPattern string) (string, error) {
+	vmBusPath := filepath.Join("/sys/bus/vmbus/devices", vmbusGUIDPattern)
+	return WaitForFileMatchingPattern(ctx, vmBusPath)
+}
+
+// WaitForFileMatchingPattern waits for a single file that matches the given path pattern and returns the full path
+// to the resulting file
+func WaitForFileMatchingPattern(ctx context.Context, pattern string) (string, error) {
```

Comment:
I love that we have this helper function now, but think it should be broken out to a utility or filesystem related package somewhere, rather than be exported from `vmbus`.

Reply:
I agree. I can make that change and update the vmbus package tests to be a small test similar to the pci package's tests.

Reply:
Done!